### PR TITLE
Updated the examples to use pytorch 1.9.0 and pytorch-lightning 1.3.5

### PR DESCRIPTION
**Pull request by KasirajanA**
Signed-off-by: KasirajanA <kasirajan.alayappan@ideas2it.com>

## What changes are proposed in this pull request?

Updated the examples to use pytorch 1.9.0 and pytorch-lightning 1.3.5

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by chauhang — APPROVED** [review]

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 1]
```diff
@@ -22,24 +22,40 @@
...
+import torchtext.datasets as td
 
 
 def get_20newsgroups(num_samples):
     categories = ["alt.atheism", "talk.religion.misc", "comp.graphics", "sci.space"]
     X, y = fetch_20newsgroups(subset="train", categories=categories, return_X_y=True)
-    return pd.DataFrame(data=X, columns=["description"]).assign(label=y).sample(n=num_samples)
+    return pd.DataFrame(
+        data=X, columns=["description"]).assign(label=y).sample(n=num_samples, replace=True
+        )
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:33]: Why we do we need `replace=True` here?

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 2]
```diff
@@ -22,24 +22,40 @@
...
 def get_ag_news(num_samples):
-    dataset_tar = download_from_url(URLS["AG_NEWS"], root=".data")
-    extracted_files = extract_archive(dataset_tar)
-    train_csv_path = list(filter(lambda x: x.endswith("train.csv"), extracted_files))[0]
-    return (
-        pd.read_csv(train_csv_path, usecols=[0, 2], names=["label", "description"])
-        .assign(label=lambda df: df["label"] - 1)  # make labels zero-based
-        .sample(n=num_samples)
-    )
+    # reading  the input
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:42]:
```suggestion
    # reading the input
```

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 3]
```diff
@@ -22,24 +22,40 @@
...
+        if fname.endswith("train.csv"):
+            train_csv_path = os.path.join(os.getcwd(), "data/AG_NEWS", fname)
+
+    df = pd.read_csv(train_csv_path)
+
+    df.columns = ["label", "title", "description"]
+    df.sample(frac=1)
+    df = df.iloc[: num_samples]
+
+    df["label"] = df.label.apply(process_label)
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:57]: Can we use method chaining here?

```
    return (
        pd.read_csv(train_csv_path, usecols=[0, 2], names=["label", "description"])
        .assign(label=lambda df: df["label"] - 1)  # make labels zero-based
        .sample(n=num_samples)
    )
```

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 4]
```diff
@@ -22,24 +22,40 @@
...
-        .sample(n=num_samples)
-    )
+    # reading  the input
+    td.AG_NEWS(root="data", split=("train", "test"))
+    extracted_files = os.listdir("data/AG_NEWS")
+
+    train_csv_path = None
+    for fname in extracted_files:
+        if fname.endswith("train.csv"):
+            train_csv_path = os.path.join(os.getcwd(), "data/AG_NEWS", fname)
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:49]: Can we simplify this par to the following?

```python
train_csv_path = "data/AG_NEWS/train.csv"
```

**Comment by harupy** [comment]
@shrinath-suresh Left some comments, thanks for the update!

File: examples/pytorch/BertNewsClassification/bert_classification.py [hunk 5]
```diff
@@ -22,24 +22,40 @@
...
     X, y = fetch_20newsgroups(subset="train", categories=categories, return_X_y=True)
-    return pd.DataFrame(data=X, columns=["description"]).assign(label=y).sample(n=num_samples)
+    return pd.DataFrame(
+        data=X, columns=["description"]).assign(label=y).sample(n=num_samples, replace=True
+        )
+
+
+def process_label(rating):
+        rating = int(rating)
+        return rating - 1
```
Comment (harupy) [inline examples/pytorch/BertNewsClassification/bert_classification.py:38]:
```suggestion
def process_label(rating):
    rating = int(rating)
    return rating - 1
```

lint

**Comment by harupy** [comment]
@shrinath-suresh Thanks for the updates! Have you locally run the example using the ag-news dataset? In CI, we

**Comment by shrinath-suresh** [comment]
> @shrinath-suresh Left some comments, thanks for the update!

@harupy Working on addressing 

> @shrinath-suresh Thanks for the updates! Have you locally run the example using the ag-news dataset? In CI, we use the `20newsgroups` that scikit-learn provides.

Y

**Comment by shrinath-suresh** [comment]
> > @shrinath-suresh Left some comments, thanks for the update!
> 
> @harupy Working on addressing
> 
> > @shrinath-suresh Thanks for the updates! Have you locally run the example using the ag-news dataset? In CI, we use the `20newsgroups` that scikit-learn provides.
> 
> Yes Haru. I did the changes and tested with AGNews. I will test it once with 20newsgroup and keep you posted.

I tested with 20newsgroup dataset and following are my observations.

1. In 20newsgroup dataset, only [4 categories](https://github.com/mlflow/mlflow/blob/ec6a16ebff6596017ab14b04554587ba892672d1/examples/pytorch/BertNewsClassification/bert_classification.py#L29) are chosen. Due to this, the sample size is around 2000 whereas the [original dataset](https://scikit-learn.org/0.19/datasets/twenty_newsgroups.html) contains around 11314 samples. Should we include all the categories ?
2.  The [default number of samples](https://github.com/mlflow/mlflow/blob/ec6a16ebff6596017ab14b04554587ba892672d1/examples/pytorch/BertNewsClassification/MLproject#L14) set for training is `15000`. It works for AGnews dataset as it contais around 20000 samples. However, for 20newsgroup, we have only 2000 samples and it returns the following error
```
Traceback (most recent call last):
  File "/home/ubuntu/mlflow/examples/pytorch/BertNewsClassification/bert_classification.py", line 403, in <module>
    dm.setup(stage="fit")
  File "/home/ubuntu/anaconda3/envs/pytorch_1.9/lib/python3.9/site-packages/pytorch_lightning/core/datamodule.py", line 384, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/ubuntu/mlflow/examples/pytorch/BertNewsClassification/bert_classification.py", line 129, in setup
    get_20newsgroups(num_samples)
  File "/home/ubuntu/mlflow/examples/pytorch/BertNewsClassification/bert_classification.py", line 33, in get_20newsgroups
    return pd.DataFrame(data=X, columns=["description"]).assign(label=y).sample(n=num_samples)
  File "/home/ubuntu/anaconda3/envs/pytorch_1.9/lib/python3.9/site-packages/pandas/core/generic.py", line 5356, in sample
    locs = rs.choice(axis_length, size=n, replace=replace, p=weights)
  File "mtrand.pyx", line 965, in numpy.random.mtrand.RandomState.choice
ValueError: Cannot take a larger sample than population when 'replace=False'
```
Shall we include all the categories in 20newsdataset and set the default num_samples to 10000 - It would work for both 20newsdataset and AGnews dataet. What do you think ?

**Comment by harupy** [comment]
Can we set the default value smaller than 2000?

**Comment by shrinath-suresh** [comment]
> Can we set the default num_samples smaller than 2000?

Sure. I have set sample size to 2000 and tested both agnews and 20newsgroup. Attached the logs

[agnews.txt](https://github.com/mlflow/mlflow/files/6800432/agnews.txt)
[20newsgroup.txt](https://github.com/mlflow/mlflow/files/6800433/20newsgroup.txt)

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**